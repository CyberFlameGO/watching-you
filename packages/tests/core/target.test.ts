import { beforeEach, describe, expect, it } from 'vitest';
import WatchingYou from '../../core';//'watching-you';
import { mousemove } from '../utils';

describe('Target', () => {
  beforeEach(() => {
    document.body.innerHTML = `
      <div class="eyes">
        <div>O</div>
      </div>
    `;
  });

  it('It should be watching mouse', async () => {
    const eyesDom = document.querySelector<HTMLElement>('.eyes');
    if (eyesDom === null) throw Error('eyesDom is not found');
    const watchingYou = new WatchingYou(eyesDom);
    watchingYou.start();
    await mousemove(100, 200);
    expect(eyesDom.style.transform).not.toBe('');
    expect(eyesDom.style.transform).not.toBe(
      'translate3d(0px,0px,0px) rotate(0deg)',
    );
    expect(eyesDom.style.transform).toMatchSnapshot();
    await mousemove(300, 300);
    expect(eyesDom.style.transform).toMatchSnapshot();
  });
});
