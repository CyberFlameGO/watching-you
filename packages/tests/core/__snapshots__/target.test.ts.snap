// Vitest Snapshot v1

exports[`Target > It should be watching mouse 1`] = `"translate3d(28.8px,40.9px,0px) rotate(144.8deg)"`;

exports[`Target > It should be watching mouse 2`] = `"translate3d(37.2px,33.4px,0px) rotate(132deg)"`;
